backend/src/news/router.py

```diff
@@ -123,22 +117,38 @@ async def news_summary(
              in the article.
     """
     response = {}
-    summary_request_payload = [
-        {
-            "role": "system",
-            "content": "你是一個新聞摘要生成機器人，請統整新聞中提及的影響及主要原因 (影響、原因各50個字，請以json格式回答 {'影響': '...', '原因': '...'})",
-        },
-        {"role": "user", "content": f"{payload.content}"},
-    ]
-
-    summarize_ai = generate_ai(summary_request_payload)
-    summary_result = summarize_ai.choices[AI.FIRST_CHOICE_INDEX].message.content
+    summary_result = openai_client.generate_summary(payload.content)
     if summary_result:
         summary_result = json.loads(summary_result)
         response["summary"] = summary_result["影響"]
         response["reason"] = summary_result["原因"]
     return response
 
+@router.post("/news_summary_custom_model")
+async def news_summary_custom_model(
+        payload: NewsSumaryCustomModelSchema, user=Depends(authenticate_user_token)
+):
+    if payload.ai_model == OpenAIConfig.model:
+        client = OpenAIClient(OpenAIConfig.api_key)
+    elif payload.ai_model == AnthropicConfig.model:
+        client = AnthropicAIClient(AnthropicConfig.api_key)
+    else:
+        raise ValueError("Invalid model specified.")
+    
+    try:
+        response = {}
+        summary_result = client.generate_summary(payload.content)
+        if summary_result:
+            summary_result = json.loads(summary_result)
+            response["summary"] = summary_result["影響"]
+            response["reason"] = summary_result["原因"]
+        return response
+    
+    except json.JSONDecodeError:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail="Error decoding JSON response.")
+    except Exception as e:
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR, detail=str(e))
```

Comment:
The business logic of the endpoint `/news_summary_custom_model` has great similarities with `/news_summary`. Were I you, I would wrap them into a function which takes a `model` parameter to avoid repetition.